src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -1092,4 +1092,12 @@ private void setupGCSClient(
   private static boolean isSuccessStatusCode(int code) {
     return code < 300 && code >= 200;
   }
+
+  /** Adds streaming ingest metadata to the StorageObjectMetadata object */
+  @Override
+  public void addStreamingIngestMetadata(
+      StorageObjectMetadata meta, String clientName, String clientKey) {
+    meta.addUserMetadata("client-name", clientName);
+    meta.addUserMetadata("client-key", clientKey);
```

Comment:
Nit, should we make `client-name` and `client-key` static final Strings somewhere and use them across the various storage clients? That way we're consistent.